src/components/pages/form-intermediate-step-page/FormIntermediateStepPage.tsx

```diff
@@ -0,0 +1,169 @@
+import { ParsedHtml } from 'components/_common/parsed-html/ParsedHtml';
+import { AnalyticsEvents, logAmplitudeEvent } from 'utils/amplitude';
+import { usePageConfig } from 'store/hooks/usePageConfig';
+
+import styles from './FormIntermediateStepPage.module.scss';
+
+export const FormIntermediateStepPage = (
+    props: FormIntermediateStepPageProps & ContentCommonProps
+) => {
+    const { data } = props;
+
+    const router = useRouter();
+    const { language } = usePageConfig();
+    const [currentPage, setCurrentPage] = React.useState(0);
+    const [selectionPath, setSelectionPath] = React.useState<number[]>([0, 0]);
+
+    const getTranslations = translator('form', language);
+
+    const buildStepObjectFromNestedStructure = (_currentPage: number) => {
+        if (_currentPage === 0) {
+            return {
+                editorial: data.editorial,
+                stepsHeadline: data.stepsHeadline,
+                steps: data.steps.map((step) => ({
+                    label: step.label,
+                    explanation: step.explanation,
+                    externalUrl: step.nextStep.external?.externalUrl,
+                })),
+            };
+        }
+        if (_currentPage === 1) {
+            const stepDetails = data.steps[selectionPath[0]].nextStep;
+            return {
+                editorial: stepDetails.next.editorial,
+                stepsHeadline: stepDetails.next.stepsHeadline,
+                steps: stepDetails.next.steps,
+            };
+        }
+    };
+
+    const backHandler = (e: React.MouseEvent) => {
+        e.preventDefault();
+        setCurrentPage(currentPage - 1);
+        const analyticsData = {
+            komponent: 'mellomsteg',
+            seksjon: 'steg-1',
+            destinasjon: router.asPath,
+            lenketekst: 'Tilbake',
+        };
+
+        logAmplitudeEvent(AnalyticsEvents.NAVIGATION, analyticsData);
+    };
+
+    const nextHandler = (
+        e: React.MouseEvent,
+        clickedPanelIndex: number,
+        label: string
+    ) => {
+        e.preventDefault();
+        const newPath = [...selectionPath];
+        newPath[currentPage] = clickedPanelIndex;
+        setSelectionPath([...newPath]);
+        gotoNextPage(newPath);
+
+        const analyticsData = {
+            komponent: 'mellomsteg',
+            seksjon: 'steg-2',
+            destinasjon: router.asPath,
+            lenketekst: label,
+        };
+
+        logAmplitudeEvent(AnalyticsEvents.NAVIGATION, analyticsData);
+    };
+
+    const gotoNextPage = (updatedPath: number[]) => {
+        // Need to drill down to find if the panel should lead
+        // to a new step or an external url
+        const currentPageStepData =
+            buildStepObjectFromNestedStructure(currentPage);
+        const nextPageToGoTo =
+            currentPageStepData.steps[updatedPath[currentPage]];
+
+        if (nextPageToGoTo.externalUrl) {
+            router.push(nextPageToGoTo.externalUrl);
```

Comment:
Bør ikke bruke next-router for eksterne lenker (og bør uansett bruke eksisterende lenke-komponenter for lenker)